### ISSUE TITLE
Can't Unlock Any Of My Four Geth-Generated Accounts, 30,000 ETH Inside

### ISSUE THREAD

**Issue by testerguy** [ethereum/go-ethereum#2908]
**I thought I should edit to include an update right at the top of the body: three of the four Geth accounts have been unlocked thanks to @DcyMatrix's suggestion below**

Requesting advice/help in solving this problem:

I created four Geth accounts in late May, and gave each a different password, which I carefully noted down (I wrote the passwords down on paper, then I entered them into the Geth command line).

Yesterday I tried to unlock the accounts, and none of the passwords worked, for any of the accounts. I'm transcribing the passwords from the same piece of paper that I initially read from when entering them into the Geth client, so it's very unlikely that this was an input error.

I've searched the internet and found these four links where people report having similar problems:

Mist Wallet password problem - https://www.reddit.com/r/ethereum/comments/4na19x/mist_wallet_password_problem/

Mist Wallet says wrong password, be careful - https://www.reddit.com/r/ethereum/comments/4jzxft/mist_wallet_says_wrong_password_be_careful/

Mist password wrong error 0.7.6 - https://github.com/ethereum/mist/issues/923

Password Assistance Needed :) - https://forum.ethereum.org/discussion/8422/password-assistance-needed

A common factor in all of these is accounts being created in May, so it may have something to do with a version of Geth that was in wide use in May. 

Someone in the discussion linked by the second URL from the top mentioned fixing their problem with Jaxx Kryptokit. I tried importing one of the accounts into the Jaxx Kryptokit wallet:

https://ethereumwallet.com/beta/index.html#dYb9

and put in the password, and it did not work, so I do believe that the password is wrong.

I found this bug report relating to both Mist and Geth: 

https://github.com/ethereum/mist/issues/182#issuecomment-198908871

I'm wondering if this could be related.

I created two new accounts today with two new passwords, and was able to unlock both with their passwords, but I did that with Geth 1.5.0-unstable and the version I was running in late May was different. 

If I lose this ETH this would set me back a few years, so I'm very anxious right now. Any help would be appreciated.

Edit, more details:

I was running Geth 1.5.0 unstable, with this snapshot: 1.5.0+986SNAPSHOT20160511110231trusty-0ubuntu1 

So this is the snapshot as of May 11th 2016

**Comment by DcyMatrix** [comment]
Had the same problem, the way I got around it then as får as I recall. Was ro type in the password in a simple text editor, confirm it was typed correctly. Then cope & paste from the editor to the Mist wallet. And it worked for me.

**Comment by testerguy** [comment]
I imported the keystore into a new machine with Mist, and none of the passwords worked, in any of the ways I tried. I'll try again later tonight/tomorrow and pay attention to use the method you mention.

**Comment by DcyMatrix** [comment]
As you I tried multiple times to type it in, I am guessing that the copy & paste method from a simple text editor fixes the problem because for some reason the keyboard input is just not always correct in the terminal.

**Comment by DcyMatrix** [comment]
Good luck! Let the rest of us know if it's a viable workaround in your case as well 👍

**Comment by testerguy** [comment]
Thanks. If I get it working I'll be sure to update everyone.

**Comment by testerguy** [comment]
I don't know how to message people on Github. Do I just mention them? @hiddentao, do you have any idea if the carriage return bug you mention here:

https://github.com/ethereum/mist/issues/182#issuecomment-198908871

could have affected passphrase entry during account creation in Geth?

@teasider and @DavidMc0, were you able to solve your locked account problems?

**Comment by fjl** [comment]
@testerguy That's definitely a possibility. I'll try to build a 
workaround/test version on Monday
that unlocks with and without the carriage return character 
appended to the password.

**Comment by testerguy** [comment]
Thank you @fjl.

**Comment by teasider** [comment]
@testerguy - I wasnt able to open that wallet. I just saw that bug you were talking about and I hope
@fjl will be able to deliver that version - it might be my problem as well (as I dont have any other passwords than the ones I've written down and use, just like you.)

Thank you for this topic - Its nice to see more options as I've used out every possible one i could find with google (Except this one with the carriage return character)

**Comment by testerguy** [comment]
What OS are you running @teasider? Do you know what version of Geth you were running when you created your passwords?

**Comment by teasider** [comment]
@testerguy I am (and was) running Windows 7 64bit.

I'm not sure what geth version I was running but the date it was created was around the end of May - and i was always up to date with whatever version was out.

**Comment by testerguy** [comment]
Ok. Also should be noted that windows uses \r\n for carriage return, whereas OSX and Linux (which I'm on) use \n

**Comment by teasider** [comment]
I also have to add that I've made around 4-5 wallets at that time, all with the same password - and its just this one wallet that doesn't seems to work.

I've never used any other passwords, I tried using a password variation tool which gives back a list of "possible typing mistakes" (one or two letters for each character) - which didn't work either (that was about 50 hours of brute forcing with that list)

So this error you just wrote about might be my last chance, Hope we figure it out for your fund's sake and mine.

**Comment by DavidMc0** [comment]
@testerguy I haven't been able to resolve my issue yet.

I'm running on Windows 10 64 bit.

I also have special characters in my passwords, sometime including "!"

I'll be away for a couple of weeks, but will try to provide any useful information once I'm back, or test any possible fixes.

Best of luck on this everyone!

**Comment by Raskal8** [comment]
The first geth account I ever created was using linux, back when gpu mining was just beginning, I used the cli command personal.newAccount("passphrase") after I mined a block I checked and my password wasn't recognized. I created a new account with personal.newAccount() and entered my password when prompted and everything worked fine. Not sure if this info helps at all

**Comment by testerguy** [comment]
@DcyMatrix the method you recommended worked for three of my four accounts!

"Had the same problem, the way I got around it then as får as I recall. Was ro type in the password in a simple text editor, confirm it was typed correctly. Then cope & paste from the editor to the Mist wallet. And it worked for me."

@teasider @DavidMc0 I recommend you try this as well.

**Comment by teasider** [comment]
Thanks, Tried it but with no luck. (one of the first things i've tried)

**Comment by DavidMc0** [comment]
thanks, and very glad to hear that this worked for 3 of your accounts!

I also tried this with Notepad in Windows 10 & no luck for me.

**Comment by testerguy** [comment]
@teasider, I recommend it once more, making sure that Mist is synced up before you do. Mist has(had?) a bug where any send error would output the 'wrong password' message, so there's a possibility that your transaction was failing for other reasons, and it was incorrectly reporting to you the cause as a 'wrong password'. It's a long shot, but you need to eliminate all of the long shot possibilities.

@DavidMc0 thanks!

**Comment by dbfrav** [comment]
Hello all - I've had this precise issue for over a month having created issue #982.  @DcyMatrix , I wish I could try your suggestion of copy/pasting from a text editor but my Ethereum Base wallet won't allow a Paste function. It won't even accept a Paste of an Ether address so I have to type in everything. I'm unable to Paste. 

As mentioned, I've been dealing with this for over a month as my Issue has now basically gone cold with no further user advice so this thread is breathing new hope for me.

**Comment by testerguy** [comment]
@dbfrav you can copy your private keys to a different wallet. All wallets have a keystore directory hich can be migrated between them. I recommend spending a couple of hours researching how to do migrations and backups of wallet keystores.

**Comment by hiddentao** [comment]
@testerguy Account creation in Geth should (as far as I'm aware) not be affected by the carriage return issue you're referring to. Plus that issue was about when you wanted to unlock the account from within Mist/Wallet.

**Comment by DcyMatrix** [comment]
@testerguy Awesome! :-D

To people having problems copy & pasting the password try and see if you
might be able to select the password in the simple text editor and then
drag the password to the password input box in your wallet. :-)

Den man. 15. aug. 2016 09.31 skrev Ramesh Nair <notifications@github.com>:

**Comment by dbfrav** [comment]
@testerguy  - I'm familiar with migration of keyfiles to different wallets and have already done so, having migrated my keyfiles to different wallets on different computers. Unfortunately, this did not solve the issue for me.

@DcyMatrix - As stated above I was not able to copy & paste my password int the wallet or even a wallet address, but thanks to your suggestion of drag and drop, I was able to do both but this did not unlock the wallet.

**Comment by shawkhawk** [comment]
I am also having a password problem with an account I created back in September 2015 from the command line. It isn't accepting any password I try. 

I've created accounts (from Ethereum-Wallet) in May and June of this year and those work fine. 

I've tried all the solutions recommended above but none of them have worked. 

I've copy/pasted the password from a plain text editor into Ethereum-Wallet.
I've tried to unlock the account using the `--unlock` flag in the command line and starting `geth` manually. 

Anything else I can try?

**Comment by testerguy** [comment]
If I may make a suggestion to Geth developers:

It would be very helpful if, during account creation, Geth displayed the password after the user has inputted it twice, and gave an option to print it. It would at the very least establish whether a password problem is due to human error or software bug. Right now, in cases like mine, there is no way to be certain whether a user made a mistake when creating their password, and are simply inputting the wrong one when attempting to decrypt, or if there was some software error during account creation.

@hiddentao thanks for the info. Any recommendations on what to do at this point? I'm fairly certain I didn't input the wrong password, as I wrote it down, twice, before entering it into the terminal. I would have had to misread the password from the piece of paper, to make the same typo, twice, in order to create the account with a password different from what I have written down.

**Comment by DcyMatrix** [comment]
To anyone having problems with the password not unlocking their wallet. Try
to see if using https://www.myetherwallet.com might be able to accept the
password.

**Comment by hiddentao** [comment]
@testerguy I'm really not sure to be honest. Based on what 
@shawkhawk it's possible that older versions of the clients were not handling all passwords correctly, e.g. especially if certain symbols were being used. But it's really hard to say at this point what the cause may be.

**Comment by teasider** [comment]
@hiddentao Could it be that somehow while creating the wallet - some more characters were put in and made my password different\invalid?

Or are we talking just about processing and decrypting the existing password?
(I'm asking since i tried unlocking using geth using mist and using myetherwallet with no success, while I ever just use 2 passwords for ethereum)

**Comment by hiddentao** [comment]
@teasider It's hard to say. But since trying it from the Geth command-line doesn't work either then It's possible something went awry during creation.

**Comment by dbfrav** [comment]
@DcyMatrix  - I did try the  https://www.myetherwallet.com suggestion to no avail. In fact I had a kind member of their staff try several attempts at their ideas to no avail. Also as mentioned above, I've tried migrating Keystore files into different wallets on different computers. There is definitely a bug, for lack of better description, within the system as although not affecting a high number of users, many of us have really been affected, especially those of us who have significant amounts of ethers basically lost now. 

As far as anyone thinking that perhaps wrong passwords were initially recorded, mistakes with passwords such as an o typed instead of a 0 (zero) or an uppercase instead of a lower case, a ! instead of a 1 (one), I even went to the extent of hiring "recovermywallet" .com initially thinking I may have made one of these errors creating my password. Dave, from RMW told me that if I gave him what I believed was my password, even with such possible mis-types, his brute force technology would be able to crack my password within minutes if not seconds. After literal weeks of trying, he said there was no hope and that the system somehow corrupted the password unlocking system. So one would really have to now believe that there is a very real (and dangerous) bug or corruption within.

I can only hope that this thread, and my own #982 will remain open and maybe someone eventually will identify and be able to help those of us affected recover our wallets and coins.

**Comment by DcyMatrix** [comment]
@dbfrav Sorry to hear it did not work.... I am having gut feeling that
tells me it might have something to do with the iso/ascii/characters of
some library somewhere in the systems affected. Possible maybe a wrong
keymap being used.

But I am not a programmer, so can't really pinpoint it myself if it even
might be the case.
Hope anyone affected and getting a solution would share it with the rest of
us.
I don't have a wallet in trouble myself anymore, but even though my Eth
supply was 'only' about 150 in the wallet it still was/is a very
significantly sum for me. And I did manage to get them send to another
wallet with the simple copy & paste method I mentioned before. Hope for you
guys this problem is pinpointed and solved. It is nerve wreaking to think
about it happening again.

**Comment by testerguy** [comment]
I concur with @DcyMatrix, it's a nerve-racking experience and it's very important that these incidents are minimized. I'd be happy to create some kind of validation feature, that ensures that the inputted passphrase works (can decrypt the account created) before notifying that the user that an account has been created, and also displays the passphrase so the user can confirm it's the same one they believed they inputted, but unfortunately I'm not familiar with golang or the Geth codebase, and I'm not sure where to start.

**Comment by dbfrav** [comment]
Yes, I agree with you both. I also am not even close to being a programmer or any other type of expert on teh inner workings of the system. I simply became interested in Ethereum and decided to buy a few hundred coins while they were around the $8.00 range as an investment. So I truly look up to you guys who have the knowledge and experience with the programming. And again, I can only hope that someone with the knowledge and experience will come along and be able to help solve this for us. Perhaps even someone who actually created the program itself? Thanks DcyMatrix and Testerguy for your efforts thus far.

**Comment by teasider** [comment]
I've experienced the exact same story as @dbfrav.
Same procedure, same results.

**Comment by ciindyprem** [comment]
wow thats must hurt

**Comment by obscuren** [comment]
Just to be sure; were the accounts created using the Ethereum-wallet or were the accounts created using the geth command line?

**Comment by dbfrav** [comment]
@obscuren  - I created my original Ethereum Base wallet (acct)  using Etherum. I then created another acct somehow involving Geth, unknowingly as I was a new user. It was after the secondary acct was created that the password no longer worked for either the main Base wallet or the secondary account. As for your second question, I upgraded to latest versions of Geth/Mist and that did not help at all.

**Comment by teasider** [comment]
@obscuren I've created all my wallets in myetherwallet and mist, this one was on mist.
i've used the same password in all of them - i dont have a different password (so it cant be it)
I've tried running ethcracker on geth to try and brute force using a password variation tool as well.

I've used the other accounts perfectly fine. Its just this one wallet.

**Comment by iFA88** [comment]
In the first time you have typed the password or pasted? Maybe you have other codepage used and  if you have special characters then its other given for password.

**Comment by teasider** [comment]
@iFA88 I have tried using the password variator with my local language as well. (just in case it was on that)
+caps lock and reverse caps lock

**Comment by iFA88** [comment]
Check the file date of your key file in the keystore.

**Comment by teasider** [comment]
@iFA88 
Saturday, May 21, 2016

Why?

**Comment by iFA88** [comment]
You have used stable or unstable geth in may ?

**Comment by teasider** [comment]
@iFA88 I wouldn't know - I've downloaded whatever was available at that time (even before, from start of may\end of april?)

What are the end results on the subject for both options(Stable\unstable)?

**Comment by iFA88** [comment]
I think you should give a try:
First of all you should backup right away your keystore.
Download the geth version around this date. From version 1.4.0 to 1.4.6. and try unlocking your accounts. Ofc you dont need sync. Use simple the command line like: geth --unlock "0 1 2 3"
Maybe.. maybe.. 50 ether worth it :)

**Comment by teasider** [comment]
Tried them all, thank you for the idea - nothing worked though.

**Comment by iFA88** [comment]
How are you created the accounts? Inside the geth console?

**Comment by iFA88** [comment]
Please come to https://gitter.im and message to me: iFA88
I would like speak you in priv.

**Comment by coltmerg420** [comment]
testerguy  i do not know if you fixet it or not but ,on file Properties did you unblock and undo the read-only box ?on your config file you are trying to put data on ?

**Comment by kevinmartinmauro** [comment]
The "!" character can occasionally cause problems, particularly on Linux, but probably other systems as well

If you're typing your password into a command window (probably not Mist, but rather a coder's terminal), you might want to add quotes around your password

**Comment by teasider** [comment]
@kevinmartinmauro 
I've tried typing in  geth --password <(echo -n $password)
but it results with "the system cannot find the file specified"
Can I use it using geth --unlock "accountaddress" ?

**Comment by testerguy** [comment]
@obscuren 

"Just to be sure; were the accounts created using the Ethereum-wallet or were the accounts created using the geth command line?"

I created all of the accounts with 1.5.0+986SNAPSHOT20160511110231trusty-0ubuntu1 (this was the latest release of Geth Unstable as of May 23rd). I updated my Geth to a more recent unstable release version) when I attempted to unlock it on August 12th.

**Comment by gdmfsob** [comment]
@teasider 

> My password contains one special character with is "!"

I was having a heck of a time with my password like a lot of you. I've been searching through everything the past day trying to see if I was just going crazy or there was some issue. 

In the end, the thing that worked for me was a different password than what I thought. So it was correctly returning an incorrect password message. 

My password also contained an !. The interesting thing is that, the password that worked, was the same password without the !. 

However, the ! was also the last character in the password. I copy and pasted from a text file when establishing the password to avoid errors. Did you do this too? I'm starting to think this may be part of the problem, that the last input didnt get copied over for us. If you haven't figured things out, try your password without the last input. 

cheers

**Comment by kevinmartinmauro** [comment]
On Linux, running the command "set +H" will allow Bash to use the exclamation point as a regular character in your terminal

**Comment by DcyMatrix** [comment]
Nice! Never heard of that before, always nice with one more command in the
toolkit. Now I just have to remember it! 😋

**Comment by teasider** [comment]
@gdmfsob, thank you for your reply.

I have tried every combination of my known passwords(with and without '!'), also tried copy-pasting from a text file to avoid mistakes.
still no luck.

**Comment by dbfrav** [comment]
Still no luck here either, after nearly a year. Anyone?

**Comment by ciindyprem** [comment]
how much are you willing to pay to unlock it?

**Comment by Romerus** [comment]
Same here, created the Wallet with geth in March 2016, it is not possible for me to open the wallet with Mist... Got the Keystore file but no access, maybe forgott the password but this whould be the first time that that ever happend

**Comment by fort9520** [comment]
This exact issue happened to me. I'm 100% sure I have the correct password. Anyone have any luck?

**Comment by Freakingcat** [comment]
Same problem happened to me. I created a Mist wallet and sent over 700 ETH there. I created a smart contract and sent 1 ETH there just to understand and learn how it works. For this I must have used the password and the contract was created successfully. When I wanted to transfer some ETH a few weeks later I got the "wrong password" error. I have tried every possible and impossible combination of what I think the password might be and also created a wordlist with over 7000 password combinations to no avail.

If anyone can help to resolve this issue, I offer 10% of the wallet content. Already tried Dave wallet recovery service but he couldnt find it.

**Comment by teasider** [comment]
@Freakingcat Same here. For all we can know there could be a bug in the wallet creation or something that only happens rarely.. I'm having the same issue exactly.

**Comment by Freakingcat** [comment]
I will move end of the year to Berlin and will try to visit some Ethereum meetups. Maybe one of the geeks there knows some advise. Annoying that Ethereum is not even acknowledging what is likely a bug or error.

**Comment by fort9520** [comment]
There has to be a bug. I wrote my password down in a notebook, and even successfully logged in to the wallet in the days prior. All happened to me when I got a new computer, transferred the wallet files, and updated to 0.8.10.

**Comment by dbfrav** [comment]
@Freakingcat  - Same here. I started an alternate post a year ago, https://github.com/ethereum/mist/issues/982 and am still trying to maintain hope.

**Comment by Freakingcat** [comment]
Maybe if a few people join together and everyone offers some % of their wallet as trophy for everyone who can solve this bug the nerds can be motivated to do something about it

**Comment by fort9520** [comment]
I would absolutely offer a %

**Comment by Freakingcat** [comment]
Lets see if enough people who suffer from the same problem can pull off together an initiative to get someone interested to find a solution to what in my opinion is definitely a bug

**Comment by holiman** [comment]
>  I created a Mist wallet and sent over 700 ETH there. I created a smart contract and sent 1 ETH there just to understand and learn how it works. For this I must have used the password and the contract was created successfully.

Could you share the transaction id? 
Have you verified that the wallet-file 'address' part corresponds to the address with 700 ETH ? 
What platform (win/unix/linux) and what version of Mist was used? 
And lastly, have you tried the password with `\n`, `\r\n` ? 

And lastly, does anyone have an account with not too much value in it, that could be shared with me along with the password? Then I could make a better stab at finding the cause.

**Comment by teasider** [comment]
@holiman If you can provide contact data I can share my locked wallet, it has around 117 ether + ETC inside, same errors. I used very specific passwords back then, also tried running password variations.

its a life changing sum for me, but trying to open it is better than leaving it locked forever.
I think I used MIST back then, it was around May 2016.
(tried using the mist version from back then already)

**Comment by holiman** [comment]
@teasider You can contact me using the PGP-key and email address listed at https://geth.ethereum.org/downloads/  (Martin Holst Swende).

**Comment by Freakingcat** [comment]
@holiman Thanks for looking into it.

Could you share the transaction id?
0x9d80658128502c05857dc44626dfb44df69AcE88

Have you verified that the wallet-file 'address' part corresponds to the address with 700 ETH ?
Sorry, I dont understand how to check this

What platform (win/unix/linux) and what version of Mist was used?
I use Mac, Mist version which was the latest in March 2016

And lastly, have you tried the password with \n, \r\n ?
I am sorry, but I dont understand this part too. As Mist wouldnt sync already for a few months despite trying it on a few other computers, I have checked the password with myetherwallet and now using Parity, which synced easily and has a password testing function. I tried every possible combination of that password. As I had used it to create a smart contract, I can exclude typo errors. Also checked that keyboard language wouldnt cause an error etc.

And lastly, does anyone have an account with not too much value in it, that could be shared with me along with the password? Then I could make a better stab at finding the cause.

**Comment by fort9520** [comment]
Any luck?

**Comment by Freakingcat** [comment]
Tried to download the latest Mist Wallet which had been released yesterday but again its stuck while syncing.

**Comment by Freakingcat** [comment]
Seems nobody has found a solution for the password bug yet

**Comment by FireAndTheVoid** [comment]
I had a similar issue, where I could not unlock my geth account with my carefully recorded password.  Luckily, I was able to figure out my issue.  Hopefully, this will help some of you.  Because my password was so long and because geth doesn't display the characters that are entered, to be on the safe side, I copied and pasted my password when creating the account.  Unfortunately, using ctrl+v doesn't work in a Linux terminal.  Instead, it set my password to an empty string.

To see if this is the case for you, start up geth in one terminal, then run "geth attach" in another.  In the second terminal, try unlocking your account with:

personal.unlockAccount("<YOUR_ACCOUNT_ADDRESS>","")

If it successfully unlocks, it will report "true".

As an alternative, (1) go to myethereumwallet.com > send ether & tokens, (2) load your keystore file, (3) type a space in the password field and then delete it (this will cause the "unlock" button to appear), (4) click unlock.

If you found this helpful, send what you think is appropriate to 0xbcfc16143f7ee5a96aceb736f823ea1747d102bd (ETH only).  Side note, this is NOT the account where I had accidentally set the password to an empty string.

**Comment by dbfrav** [comment]
@FireAndTheVoid  - Thank you for this. This sounds very hopeful. I'm going to put your advice to work this week and will report back here with my results. I will absolutely send significant ethers to you if successful.   dbf -

**Comment by Freakingcat** [comment]
@FireAndTheVoid I tried it with MyEtherwallet and with Parity....unfortunately it didnt work. Thanks for your advise

**Comment by FireAndTheVoid** [comment]
@Freakingcat  I would still like to help.  Some details might help, though.

1.  I saw in an above post that you are using a Mac.  What version of OSX were you running when you installed Mist?  Have you upgraded to a newer version of OSX since then?

2.  You also mentioned that you were using the latest version of Mist in the March 2016 time frame.  In March 2016, it looks like the latest version of Mist was Wallet 0.5.2 (Beta 10) (I'm referencing https://github.com/ethereum/mist/releases?after=0.7.1).  Does this sound like the version you used?

3.  When you installed Mist, did you download the pre-built version or did you download the source and build it yourself?

4.  Did you upgrade the Mist wallet software between the time you successfully executed a smart contract and the time you first received the "wrong password" error?  If so, what version of Mist did you upgrade to?

5.  Roughly how long is your password?  What special characters are included?  Important: I don't need the actual password, just an approximate length and the special characters, i.e. #$?!, in no particular order.

If you would rather not discuss the details in this forum, send me an e-mail at fir3andth3void@gmail.com.

Also, although I've never used Mist myself, it should't be necessary to sync with the network in order to check if your password works.  At least with geth, I am certain that it isn't necessary.

**Comment by Freakingcat** [comment]
Thanks a lot @FireAndTheVoid - I will send you an email in the next hours!

**Comment by dbfrav** [comment]
@FireAndTheVoid , this didn't work for me either, but thanks for your efforts.

**Comment by FireAndTheVoid** [comment]
@dbfrav If you want some help trying to unlock your account, send me an e-mail at fir3andth3void@gmail.com.  I'm still working with @Freakingcat to unlock his account, but I was successful in helping another person in the forums to get into theirs.

**Comment by ethtester** [comment]
@FireAndTheVoid  what tools are you using for password testing?  ethcrack, hashcat or JTR?  if so what is your hash rate

**Comment by FireAndTheVoid** [comment]
@ethtester I am a software developer and I have mostly developed my own tools.  My tools are based on web3j (https://github.com/web3j/web3j) which uses the lambdaWorks-OSS Google Group's implementation of scrypt (https://github.com/wg/scrypt).  On an typical 2-3 GHz Intel CPU, it completes a hash in around 0.6-0.7 seconds.  I run it multi-threaded on one or more of my server-grade workstation computers.  Currently, the number of threads I run is limited by the amount of memory on the systems (I am not CPU limited).  With each instance consuming around 6+ GB of memory, my workstations can only run around 6-7 instances simultaneously.  I'm considering buying more RAM, which would allow me to process double or triple the number of password combinations in the same amount of time.  But with my current setup, on a single workstation, I am processing around 10 passwords a second.

**Comment by ethtester** [comment]
I'm using hashcat 3.6.0 with a quad core CPU, seeing about 10/12 H/s.  On another box with 2 Xeon processors with total 16 cores I see 25 H/s.

**Comment by FireAndTheVoid** [comment]
Wow, I will have to look into hashcat. You aren't memory limited on the Xeon box?

**Comment by ethtester** [comment]
According to a hashcat dev each iteration uses 250MB of memory.  This is why we cannot crack on Scrypt on GPU's.  The best GPU's today only have about 16GB of ram, not enough to use all the 2000+ GPU cores...   So my workstations with 32GB are not memory limited.

**Comment by evertonfraga** [comment]
Cross-referencing https://github.com/ethereum/mist/issues/3513.

**Comment by anormore** [comment]
Hi there, like you I'm locked out. I've been using Hashcat a lot recently, but there is actually little point at the moment, please see this guys thread:

https://hashcat.net/forum/thread-7181-post-38590.html#pid38590

Essentially, we need to determine in the Eth wallet is being generated incorrectly. No amount of GPU cracking in the world will guess it correct if the hash value of the wallet is incorrect! Good luck, stay with us.

**Comment by ethtester** [comment]
arnormore, are you having a password issue with a pre-sale wallet or a geth/mist wallet?

**Comment by jeycreative** [comment]
Hello world, I'm joining the club too.
Here is my data : Several wallets created on OS X in march 2016 using the ether wallet app.
Some passwords were using "!" at the end too. might be a strong clue there.
Have recovered one of 4 wallet using one of my noted password 10 char in low caps without "!" at the end. For the others I'm using hashcat but it will take looooong time...
Have anyone used a password using a double exclamation point at the end (!!) and might suffer the same problem ? it might help to know. Good Luck & computing to you all.

**Comment by evertonfraga** [comment]
@MatrixManAtYrService it does make sense, but I'd take a slightly different approach in order to save time/efforts:

1. first, try to reproduce the issue manually. Make sure that a given password does not work at N version so you can write your script.
1. instead of run the tests for each commit, why not try with each version? Once you have a potential ∆ you can talk about commits.

**Comment by bigvito19** [comment]
Do yo mind if I give it a try?

**Comment by JoePuffinGoat** [comment]
I tried many different methods to recover my Geth account with no success. My final at was attempt was successful and so simple I couldn't believe I made this mistake.

1. Using Window Geth CLI
1. enter `geth account unlock 0`
1. then simply enter `Ctrl+v`.

I was only able to get this to work in the Geth CLI. An empty Ctrl+v didn't seem to work in myetherwallet, though.

**Comment by adamschmideg** [comment]
Regretfully, it's not possible to solve this issue in a technical way. It was opened a long time ago, this is not a single issue anymore.
